web/src/Legal.tsx

```diff
@@ -0,0 +1,181 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import Container from "@cloudscape-design/components/container";
+import Header from "@cloudscape-design/components/header";
+import Table from "@cloudscape-design/components/table";
+import Link from "@cloudscape-design/components/link";
+import SpaceBetween from "@cloudscape-design/components/space-between";
+
+import { useEffect, useState } from "react";
+
+type Package = {
+  package_name: string;
+  version: string;
+  recipe_name: string;
+  license: string;
+};
+
+function parse_mainifest(text: string) {
+  let packages: Package[] = [];
+
+  // The content of `text` looks something like this:
+  //
+  // PACKAGE NAME: tacd
+  // PACKAGE VERSION: 1.0.0
+  // RECIPE NAME: tacd
+  // LICENSE: GPL-2.0-or-later
+  //
+  // PACKAGE NAME: tacd-webinterface
+  // ...
+
+  for (var group of text.split("\n\n")) {
+    let pkg: Package = {
+      package_name: "",
+      version: "",
+      recipe_name: "",
+      license: "",
+    };
+
+    for (var line of group.split("\n")) {
+      if (line.startsWith("PACKAGE NAME: ")) {
+        pkg.package_name = line.replace("PACKAGE NAME: ", "");
+      }
+      if (line.startsWith("PACKAGE VERSION: ")) {
+        pkg.version = line.replace("PACKAGE VERSION: ", "");
+      }
+      if (line.startsWith("RECIPE NAME: ")) {
+        pkg.recipe_name = line.replace("RECIPE NAME: ", "");
+      }
+      if (line.startsWith("LICENSE: ")) {
+        pkg.license = line.replace("LICENSE: ", "");
+      }
+    }
+
+    if (pkg.package_name && pkg.version && pkg.recipe_name && pkg.license) {
+      packages.push(pkg);
+    }
+  }
+
+  return packages;
+}
+
+function PackageList() {
+  const [packages, setPackages] = useState<Package[]>();
+
+  useEffect(() => {
+    fetch("/docs/legal/license.manifest").then((response) => {
+      if (response.ok) {
+        response.text().then((text) => setPackages(parse_mainifest(text)));
+      }
+    });
+  }, []);
+
+  return (
+    <Table
+      header={
+        <Header
+          variant="h3"
+          description="Software packages used on this LXA TAC"
+        >
+          Packages
+        </Header>
+      }
+      columnDefinitions={[
+        {
+          id: "package_name",
+          header: "Package Name",
+          cell: (p) => p.package_name,
+        },
+        {
+          id: "version",
+          header: "Version",
+          cell: (p) => p.version,
+        },
+        {
+          id: "recipe_name",
+          header: "Recipe Name",
+          cell: (p) => p.recipe_name,
+        },
+        {
+          id: "license",
+          header: "License",
+          cell: (p) => (
+            <Link href={"/docs/legal/files/" + p.recipe_name}>{p.license}</Link>
```

Comment:
I first thought that we could just set the mime type in the link on the front end, but we also generate a file listing. So you're approach is probably a good abstraction.